### PR TITLE
Bump msal4j version & add regional STS support

### PR DESCRIPTION
**Pull request by jianghaolu**

### CODE REVIEW

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/RegionalAuthority.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// Licensed under the MIT License.
+
+package com.azure.identity;
+
+import com.azure.core.util.ExpandableStringEnum;
+
+/**
+ * Defines currently available regional authorities, or "AutoDiscoverRegion" to auto-detect the region.
+ */
+public class RegionalAuthority extends ExpandableStringEnum<RegionalAuthority> {
```
Comment (alzimmermsft) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/RegionalAuthority.java:11]: nit: `final`

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/RegionalAuthority.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    public static final RegionalAuthority AUTO_DISCOVER_REGION = fromString("AutoDiscoverRegion");
```
Comment (alzimmermsft) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/RegionalAuthority.java:12]: These will need Javadocs

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientOptions.java [hunk 3]
```diff
@@ -22,6 +23,7 @@
  */
 public final class IdentityClientOptions {
     private static final int MAX_RETRY_DEFAULT_LIMIT = 3;
+    private static final String AZURE_REGIONAL_AUTHORITY_NAME = "AZURE_REGIONAL_AUTHORITY_NAME";
```
Comment (alzimmermsft) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientOptions.java:26]: Should `AZURE_REGIONAL_AUTHORITY_NAME` be added to `Configuration`?

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/RegionalAuthority.java (hunk 2 above)
Comment (jianghaolu) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/RegionalAuthority.java:12]: Can we skip them? They are sort of self-explanatory.

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientOptions.java (hunk 3 above)
Comment (jianghaolu) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientOptions.java:26]: Are we supposed to add all service environment variables to `azure-core`?

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/RegionalAuthority.java (hunk 2 above)
Comment (jianghaolu) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/RegionalAuthority.java:18]: added docs

**Review by alzimmermsft — APPROVED** [review]

File: sdk/core/azure-core/src/main/java/com/azure/core/util/Configuration.java [hunk 4]
```diff
@@ -88,6 +88,11 @@
      */
     public static final String PROPERTY_AZURE_IDENTITY_DISABLE_CP1 = "AZURE_IDENTITY_DISABLE_CP1";
 
+    /**
+     * Name of Azure AAD regional authority.
+     */
+    public static final String PROPERTY_AZURE_REGIONAL_AUTHORITY_NAME = "AZURE_REGIONAL_AUTHORITY_NAME";
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/Configuration.java:94]: @jianghaolu This property is not included in `DEFAULT_CONFIGURATIONS` array. All other properties are included. Is there a reason we don't want this to be included?

**Pull request merged**